docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,598 @@
 title: Adding search with Algolia
 ---
 
-This is a stub. Help our community expand it.
+Once you've added some content to your site, you'll want to make it's easy for your visitor to find what they're looking for. This guide will run you through the process of setting up a custom search experience powered by [Algolia](https://www.algolia.com).
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Before, we begin I should mention that if you're looking to add search to a documentation site, then you can let Algolia handle most of the steps outlined below for you by using their excellent [Docsearch](https://community.algolia.com/docsearch) functionality. For other types of sites and more fine-grained control over exactly what data should be indexed, read on.
+
+## Backend
+
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. With `react-instantsearch` Algolia provides an extensive library of off-the-shelf React components that we can simply import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https://github.com/motdotla/dotenv) while you're at it. We're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
+
+```sh
+yarn add gatsby-plugin-algolia react-instantsearch-dom dotenv
```

Comment:
also add `algoliasearch` to prevent depending on transitive dependencies

Reply:
Just added that. Didn't know it introduces a transitive dependence.